public/app/features/explore/utils/ResultProcessor.ts

```diff
@@ -128,8 +127,25 @@ export function isTimeSeries(frame: DataFrame, datasource?: string): boolean {
   return false;
 }
 
+/* In Explore table, we don't want to show time-series.
+ * EXCEPTION: For Prometheus, we want to show time-series when they are result of instant queries.
+ */
+function shouldShowInTable(frame: DataFrame, datasource?: string) {
+  // DataFrames with exactly 2 fields, when first field.type === time are time-series.
+  if (frame.fields.length === 2 && frame.fields[0].type === FieldType.time) {
+    // Hack: Prometheus instant vectors's value has always length === 1.
+    // https://prometheus.io/docs/prometheus/latest/querying/api/#instant-vectors
+    if (datasource && datasource === 'prometheus' && frame.length === 1) {
+      return true;
+    }
+    return false;
+  }
```

Comment:
I would in the prometheus data source  set a meta field on the time series results that should be hidden in explore. Then we can always show any data in the table (that does not have a hideExploreTable property on them)